### PR TITLE
add prune_with_map_projection

### PR DESCRIPTION
**Pull request by hannes14**
@ederc

### CODE REVIEW

**Comment by ederc** [comment]
@RafaelDavidMohr Could you please have a look if these changes include the information you need?

File: src/module/module.jl [hunk 1]
```diff
@@ -407,6 +408,21 @@ function prune_with_map(I::smodule{spoly{T}}) where T <: Nemo.FieldElem
    return smodule{spoly{T}}(R, r),smatrix{spoly{T}}(R,TT_ptr)
 end
 
+@doc raw"""
+    prune_with_map_projection(id::smodule{spoly{T}}) where T <: Nemo.FieldElem
+
+Returns the module R minimally embedded in a free module such that the
```
Comment (fingolfin) [inline src/module/module.jl:414]: Should be "Return", not "Returns" (also in `prune_with_map` above). And I guess `R` should be code?
```suggestion
Return the module `R` minimally embedded in a free module such that the
```

File: src/module/module.jl [hunk 2]
```diff
@@ -407,6 +408,21 @@ function prune_with_map(I::smodule{spoly{T}}) where T <: Nemo.FieldElem
...
+corresponding factor modules are isomorphic,
```
Comment (fingolfin) [inline src/module/module.jl]: I don't understand this docstring: There is one argument `id`; it is not stated how it related to `R`. And then the next line, is that transformation matrix meant to be another return value?

How about this (assuming I guessed the meaning right, which I probably didn't):
```suggestion
Return a module `R`, a transformation matrix `M` and a projection map `p`
satisfying the following properties: `R` is minimally embedded in a free
module `parent(R)` such that the quotient `parent(R)/R` is isomorphic to `id`,
with `M` describing this isomorphism, and `p` mapping `gen(id, i)` to
`gen(id, p(i))` 
```
I am still rather confused by the usage of `gen(i)` (what is `i`? An index ranging over `1 .. ngens(id)` maybe?!?), and why `p[i]` -- is `p` a vector, or a function, or what?

**Comment by RafaelDavidMohr** [comment]
If @fingolfin's documentation update above is correct, then yes, this is what I need. Essentially I just need the map `parent(R)/R -> id`

**Review by ederc — APPROVED** [review]

**Pull request merged**